src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -950,22 +942,26 @@ async def start_grpc_server(
     return server
 
 
-async def run_grpc_server(
-    engine: AsyncLLMEngine,
-    args: argparse.Namespace,
-    *,
-    disable_log_stats: bool,
-) -> None:
-    assert args is not None
+async def run_grpc_server(args: argparse.Namespace) -> None:
+    from vllm.entrypoints.openai.api_server import AsyncEngineRPCClient
 
-    server = await start_grpc_server(engine, args)
+    rpc_server_port = vllm.envs.VLLM_RPC_PORT
+    logger.info("gRPC server connecting to async engine on port %s", rpc_server_port)
+    async_engine_client = AsyncEngineRPCClient(rpc_server_port)
+    await async_engine_client.setup()
+    server = await start_grpc_server(
+        async_engine_client,
+        args,
+    )
 
     try:
         while True:
             await asyncio.sleep(10)
-            if not disable_log_stats:
-                await engine.do_log_stats()
     except asyncio.CancelledError:
         print("Gracefully stopping gRPC server")  # noqa: T201
+    except Exception:
+        logger.exception("Engine is dead")
+        # FIXME: handle this case?
+    finally:
         await server.stop(30)  # TODO configurable grace
         await server.wait_for_termination()
```

Comment:
Again not critical but seems like there should be a cleaner way to do this

Reply:
The `except Exception` clause was a leftover from an older revision, I'll clean it up in the next push

Reply:
Cleaned it up